lib/src/scheduler_creator.rs

```diff
@@ -0,0 +1,25 @@
+use petgraph::Graph;
+
+use crate::{
+    fixed_priority_scheduler::FixedPriorityScheduler, graph_extension::NodeData,
+    processor::ProcessorBase, scheduler::DAGSchedulerBase,
+};
+
+pub enum SchedulerType {
+    FixedPriorityScheduler,
+}
+
+pub fn create_scheduler<T>(
+    scheduler_type: SchedulerType,
+    dag: &mut Graph<NodeData, i32>,
+    processor: &T,
+) -> Box<impl DAGSchedulerBase<T> + 'static>
+where
+    T: ProcessorBase + Clone + 'static,
+{
+    match scheduler_type {
+        SchedulerType::FixedPriorityScheduler => {
+            Box::new(FixedPriorityScheduler::new(dag, processor))
+        }
+    }
```

Comment:
Add `_`

Reply:
Since the decision is `enum SchedulerType`, the following warning will appear if `_` is entered.
This is because the input is always `SchedulerType`.

```
unreachable pattern
`#[warn(unreachable_patterns)]` on by default
```

Reply:
OK